cpp/csp/adapters/websocket/WebsocketEndpoint.h

```diff
@@ -1,84 +1,367 @@
 #ifndef _IN_CSP_ADAPTERS_WEBSOCKETS_ENDPOINT_H
 #define _IN_CSP_ADAPTERS_WEBSOCKETS_ENDPOINT_H
 
-// need a base -> TLS, base -> No TLS
-#include <csp/adapters/websocket/ClientInputAdapter.h>
-#include <websocketpp/config/asio_client.hpp>
-#include <websocketpp/config/asio_no_tls_client.hpp>
-#include <websocketpp/client.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/ssl.hpp>
+#include <boost/beast/websocket.hpp>
+#include <boost/asio/strand.hpp>
+
+#include <csp/engine/Dictionary.h>
+#include <csp/core/Exception.h>
+
+#include <cstdlib>
 #include <functional>
+#include <iostream>
+#include <string>
 
 namespace csp::adapters::websocket {
+using namespace csp;
 
-using on_message_cb = std::function<void(std::string)>;
-using on_send_fail_cb = std::function<void(const std::string&)>;
-using void_cb = std::function<void()>;
-
-using message_ptr = websocketpp::config::core_client::message_type::ptr;
+namespace beast = boost::beast;         // from <boost/beast.hpp>
+namespace http = beast::http;           // from <boost/beast/http.hpp>
+namespace net = boost::asio;            // from <boost/asio.hpp>
+namespace ssl = boost::asio::ssl;       // from <boost/asio/ssl.hpp>
+namespace websocket = beast::websocket; // from <boost/beast/websocket.hpp>
+using tcp = boost::asio::ip::tcp;       // from <boost/asio/ip/tcp.hpp>
 
+using string_cb = std::function<void(const std::string&)>;
```

Comment:
this should be changed to const char * as well to avoid the conversion

Reply:
changed to pass void* all the way through